test/path_matcher_test.cc

```diff
@@ -819,6 +823,37 @@ TEST_F(PathMatcherTest, VariableBindingsWithQueryParamsEncoding) {
             bindings);
 }
 
+TEST_F(PathMatcherTest, QueryParameterNotUnescapePlus) {
+  MethodInfo* a = AddGetPath("/a");
+  Build();
+
+  EXPECT_NE(nullptr, a);
+
+  Bindings bindings;
+  EXPECT_EQ(LookupWithParams("GET", "/a", "x=Hello+world&y=%2B+%20", &bindings), a);
```

Comment:
Could you comment the mapping of URL encoding here for readability?

Reply:
Done